### PR TITLE
Regex and Formatting - url domain forwardslash

### PR DESCRIPTION
**Pull request by bakatzir**
## Status
In Progress

## Related Issues
fixes: https://github.com/demisto/etc/issues/19449

## Description
Support http: and http:\\ in both domain and url regexes and formatting scripts.

## Screenshots
in referenced issue.

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [x] Documentation - no relevant docs.
- [ ] Code Review

## Technical writer review
Mention and link to the files that require a technical writer review.
- [ ] CHANGELOGS:
https://github.com/demisto/content/blob/e486c13df3251e5cfb9637fcb4a94e321cf3b1f7/Misc/reputation-domain_CHANGELOG.md
https://github.com/demisto/content/blob/e486c13df3251e5cfb9637fcb4a94e321cf3b1f7/Misc/reputation-url_CHANGELOG.md
https://github.com/demisto/content/blob/e486c13df3251e5cfb9637fcb4a94e321cf3b1f7/Scripts/ExtractDomainFromUrlFormat/CHANGELOG.md
https://github.com/demisto/content/blob/e486c13df3251e5cfb9637fcb4a94e321cf3b1f7/Scripts/script-UnEscapeURLs_CHANGELOG.md

### CODE REVIEW

File: Scripts/ExtractDomainFromUrlFormat/ExtractDomainFromUrlFormat.py [hunk 1]
```diff
@@ -37,7 +37,12 @@ def proofpoint_get_original_url(safe_url):
...
 
 def unescape_url(escaped_url):
-    url = escaped_url.lower().replace('[.]', '.').replace('hxxp', 'http').replace('&amp;', '&')
+    # Normalize: 1) [.] --> . 2) hxxp --> http 3) &amp --> & 4) http:\\ --> http://
+    url = escaped_url.lower().replace('[.]', '.').replace('hxxp', 'http').replace('&amp;', '&')\
+        .replace('http:\\\\', 'http://')
+    # Normalize the URL with http prefix
+    if url.find('http:') > 0 and url.find('http://') != 0:
+        url = url.replace('http:', 'http://')
     if url.find('http') != 0 and url.find('ftp') != 0:
```
Comment (glicht) [inline Scripts/ExtractDomainFromUrlFormat/ExtractDomainFromUrlFormat.py]: Add unit tests please

File: TestPlaybooks/playbook-UnEscapeURLs-Test.yml [hunk 2]
```diff
@@ -593,12 +593,12 @@ tasks:
       - "23"
     scriptarguments:
       input:
-        simple: https://urldefense.com/v3/__http:/www.nintendo.com__;!83Wvu-Cv-B_8k64!-XmeoAPIhKFgj4X-lVVmbQXWouodrfqiYRhDEj7kMbX6Kx_EZpL_Nh1iKyoLTt4QjpI$
+        simple: http:\\example.com
```
Comment (glicht) [inline TestPlaybooks/playbook-UnEscapeURLs-Test.yml]: seems you deleted the previous test

File: Scripts/ExtractDomainFromUrlFormat/ExtractDomainFromUrlFormat_test.py [hunk 3]
```diff
@@ -15,3 +16,12 @@
 def test_extract_domain(input, domain):
     res = extract_domain(input)
     assert res == domain
+
+
+@pytest.mark.parametrize('input,url', [
+                        ('http:example.com', 'http://example.com'),
+                        ('http:\\\\example.com', 'http://example.com'),
+                        ])
```
Comment (glicht) [inline Scripts/ExtractDomainFromUrlFormat/ExtractDomainFromUrlFormat_test.py]: Add also these urls to the above test: `test_extract_domain` so we are sure we are covered full flow.

File: TestPlaybooks/playbook-UnEscapeURLs-Test.yml [hunk 4]
```diff
@@ -630,7 +631,73 @@ tasks:
...
+      scriptName: UnEscapeURLs
+      type: regular
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "25"
+    scriptarguments:
+      input:
+        simple: http:\\example.com
```
Comment (glicht) [inline TestPlaybooks/playbook-UnEscapeURLs-Test.yml]: Add also a test for no slash case: `http:example.com`

Comment (bakatzir) [reply TestPlaybooks/playbook-UnEscapeURLs-Test.yml]: added the specified testcase

**Comment by bakatzir** [comment]
@glicht 
We have tests for the domain regex in `reputations.json Test`

**Comment by Itay4** [comment]
did we do performance tests on these regexes ?

**Comment by bakatzir** [comment]
@Itay4 I haven't.

**Comment by glicht** [comment]
@bakatzir from what I see in `reputations.json Test` playbook we are only testing urls with the specified new url formats and not domain extraction.

**Comment by bakatzir** [comment]
@Itay4 performance tests are running. one build with my changes, second with master.
https://circleci.com/gh/demisto/server/tree/regex_performence_test
will update with the results.

**Comment by bakatzir** [comment]
@Itay4 @glicht 
performence test with the regex build:
https://circleci.com/gh/demisto/server/52293

**Comment by kirbles19** [comment]
@bakatzir Done.

**Pull request merged**